src/features/game/components/CloseablePanel.tsx

```diff
@@ -13,13 +13,33 @@ export interface PanelTabs {
   name: string;
 }
 
+/**
+ * The base props for closeable panels that may have a close button.
+ * @param showCloseButton The item resources requirements.
+ * @param onClose The SFL requirements.
+ */
+interface HasCloseButtonProps {
```

Comment:
I see what you're trying to do here, however, this is a pretty confusing implementation. As these reuseable components grow we need to really keep in consideration whether or not they're becoming too flexible (confusing). For example, it might make sense that this panel should only be used when looking for something with a close button especially since its called the `CloseablePanel`.

Either way, I think an easier way to achieve what you were trying to do would be to have th `onClose` prop be optional and then we can conditionally render the close button based on the existence of this prop.

Reply:
Addressed